core/src/main/resources/org/apache/spark/ui/static/stagepage.js

```diff
@@ -0,0 +1,906 @@
+                        },
+                        {
+                            data : function (row, type) {
+                                return typeof row.memoryBytesSpilled != 'undefined' ? formatBytes(row.memoryBytesSpilled, type) : "";
+                            }
+                        },
+                        {
+                            data : function (row, type) {
+                                return typeof row.diskBytesSpilled != 'undefined' ? formatBytes(row.diskBytesSpilled, type) : "";
+                            }
+                        }
+                    ],
+                    "order": [[0, "asc"]]
+                }
+                var executorSummaryTableSelector =
+                  $(executorSummaryTable).DataTable(executor_summary_conf);
+                $('#parent-container [data-toggle="tooltip"]').tooltip();
+
+                // building accumulator update table
+                var accumulatorTable = "#accumulator-table";
+                var accumulator_conf = {
+                    "data": accumulator_table,
+                    "columns": [
+                        {data : "id"},
+                        {data : "name"},
+                        {data : "value"}
+                    ],
+                    "paging": false,
+                    "searching": false,
+                    "order": [[0, "asc"]]
+                }
+                $(accumulatorTable).DataTable(accumulator_conf);
+
+                // building tasks table that uses server side functionality
+                var taskTable = "#active-tasks-table";
+                var task_conf = {
+                    "serverSide": true,
+                    "paging": true,
+                    "info": true,
+                    "processing": true,
+                    "lengthMenu": [[20, 40, 60, 100, responseBody.numTasks], [20, 40, 60, 100, "All"]],
+                    "ajax": {
+                        "url": stageEndPoint(appId) + "/" + stageAttemptId + "/taskTable",
+                        "data": {
+                            "numTasks": responseBody.numTasks
+                        },
+                        "dataSrc": function ( jsons ) {
+                            var jsonStr = JSON.stringify(jsons);
+                            var tasksToShow = JSON.parse(jsonStr);
+                            return tasksToShow.aaData;
+                        }
+                    },
+                    "columns": [
+                        {data: function (row, type) {
+                            return type !== 'display' ? (isNaN(row.index) ? 0 : row.index ) : row.index;
+                            },
+                            name: "Index"
+                        },
+                        {data : "taskId", name: "ID"},
+                        {data : "attempt", name: "Attempt"},
+                        {data : "status", name: "Status"},
+                        {data : "taskLocality", name: "Locality Level"},
```

Comment:
nit, could `taskLocality` be sorted: process, node, rack, and then any? where process is 0, and any is 4?

Reply:
As discussed, I am using existing sorting API's so I have to explicitly write code to sort it the way described above. If it is nit for now, can do it as part of another PR.